src/mlpack/bindings/python/print_input_processing.hpp

```diff
@@ -91,15 +189,70 @@ void PrintInputProcessing(
     else
       std::cout << name;
     std::cout << ")" << std::endl;
-    std::cout << prefix << "CLI.SetPassed(<const string> '" << d.name << "')"
+    std::cout << prefix << "  CLI.SetPassed(<const string> '" << d.name << "')"
         << std::endl;
+
+    if (def == "False")
+    {
+        std::cout << "  else:" << std::endl;
+        std::cout << "    raise TypeError(" <<"\"'"<< name
+            <<"' must have type \'bool'!\")" << std::endl;
+    }
+    else if (GetCythonType<T>(d) == "string")
+    {
+      std::cout << "  else:" << std::endl;
+      std::cout << "    raise TypeError(" <<"\"'"<< name
+          <<"' must have type \'str'!\")" << std::endl;
+    }
+    else if (GetCythonType<T>(d) == "vector[string]"
+      ||GetCythonType<T>(d) == "vector[int]")
+    {
+      std::cout << "  else:" << std::endl;
+      std::cout << "    raise TypeError(" <<"\"'"<< name
+          <<"' must have type \'list'!\")" << std::endl;
+    }
+    else if (GetCythonType<T>(d) == "double")
+    {
+      std::cout << "  else:" << std::endl;
+      std::cout << "    raise TypeError(" <<"\"'"<< name
+          <<"' must have type \'float'!\")" << std::endl;
+    }
+    else if (GetCythonType<T>(d) == "int")
+    {
+      std::cout << "  else:" << std::endl;
+      std::cout << "    raise TypeError(" <<"\"'"<< name
+          <<"' must have type \'int'!\")" << std::endl;
+    }
   }
   std::cout << std::endl; // Extra line is to clear up the code a bit.
 }
 
 /**
  * Print input processing for a matrix type.
  */
+
+/**
+ * This gives us code like:
+ *
+ * # Detect if the parameter was passed; set if so.
+ * if param_name is not None:
+ *   if isinstance(param_name, np.ndarray) or isinstance(param_name,
+ *       pd.DataFrame) or isinstance(param_name, pd.Series):
```

Comment:
I think we can accept `list` here too (really any "arraylike" works).  Actually I would be okay with encoding the check in `to_matrix()` directly, since `to_matrix()` is the method that actually controls whether or not the type can be accepted.